src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,992 @@
+
+            if response_kubectl_get_events.returncode != 0:
+                telemetry.set_exception(exception=error_kubectl_get_events.decode("ascii"), fault_type=consts.Kubectl_Get_Events_Failed, summary='Error while doing kubectl get events')
+                logger.warning("Error while doing kubectl get events")
+
+            # Converting output obtained in json format and fetching the clusterconnect-agent feature
+            events_json = json.loads(output_kubectl_get_events)
+
+            # Path to add the azure-arc events
+            event_logs_path = os.path.join(filepath_with_timestamp, "Arc_Agents_Events.txt")
+            with open(event_logs_path, 'w+') as event_log:
+
+                # Adding all the individual events
+                for events in events_json["items"]:
+                        event_log.write(str(events) + "\n")
+
+            return consts.Diagnostic_Check_Passed, storage_space_available
+
+    # For handling storage or OS exception that may occur during the execution
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            storage_space_available = False
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+        else:
+            logger.warning("An exception has occured while trying to fetch the events occured in azure-arc namespace from the cluster. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Arc_Agents_Events_Logger_Fault_Type, summary="Error occured in arc agents events logger")
+            diagnoser_output.append("An exception has occured while trying to fetch the events occured in azure-arc namespace from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while trying to fetch the events occured in azure-arc namespace from the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Arc_Agents_Events_Logger_Fault_Type, summary="Error occured in arc agents events logger")
+        diagnoser_output.append("An exception has occured while trying to fetch the events occured in azure-arc namespace from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return consts.Diagnostic_Check_Failed, storage_space_available
+
+
+def retrieve_deployments_logs(appv1_api_instance, filepath_with_timestamp, storage_space_available):
+
+    global diagnoser_output
+    try:
+
+        if storage_space_available:
+
+            # Creating new Deployment Logs folder in the given timestamp folder
+            deployments_path = os.path.join(filepath_with_timestamp, "Deployment_logs")
```

Comment:
nit: rename to "arc_deployment_logs_path"

and file name to "arc_deployment_logs"